components/usage/pkg/server/server.go

```diff
@@ -151,12 +151,12 @@ func Start(cfg Config) error {
 	return nil
 }
 
-func registerGRPCServices(srv *baseserver.Server, conn *gorm.DB, stripeClient *stripe.Client, reportGenerator *apiv1.ReportGenerator, contentSvc contentservice.Interface, billInstancesAfter time.Time) error {
+func registerGRPCServices(srv *baseserver.Server, conn *gorm.DB, stripeClient *stripe.Client, reportGenerator *apiv1.ReportGenerator, contentSvc contentservice.Interface, billInstancesAfter time.Time, selfConnection grpc.ClientConnInterface) error {
```

Comment:
Here, I'd change the interface to instead of taking a `selfConnection`, make it take an instance of `v1.UsageServiceClientInterface` (unsure if that's the exact type). This means that you no longer deal with how to establish the conenction, you just receive a client.

Reply:
This will be moved to the PR that tackles the use of the RPC.